server/sonar-server/src/main/java/org/sonar/server/qualityprofile/QProfileResetImpl.java

```diff
@@ -0,0 +1,164 @@
+ */
+package org.sonar.server.qualityprofile;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ArrayListMultimap;
+import com.google.common.collect.ListMultimap;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Sets;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import org.sonar.api.profiles.ProfileDefinition;
+import org.sonar.api.profiles.RulesProfile;
+import org.sonar.api.rule.RuleKey;
+import org.sonar.api.rules.ActiveRule;
+import org.sonar.api.rules.ActiveRuleParam;
+import org.sonar.api.server.ServerSide;
+import org.sonar.api.utils.ValidationMessages;
+import org.sonar.db.DbClient;
+import org.sonar.db.DbSession;
+import org.sonar.db.organization.OrganizationDto;
+import org.sonar.db.qualityprofile.ActiveRuleDto;
+import org.sonar.db.qualityprofile.ActiveRuleKey;
+import org.sonar.db.qualityprofile.QualityProfileDto;
+import org.sonar.db.rule.RuleParamDto;
+import org.sonar.server.exceptions.BadRequestException;
+import org.sonar.server.qualityprofile.index.ActiveRuleIndexer;
+
+import static org.sonar.server.ws.WsUtils.checkRequest;
+
+@ServerSide
+public class QProfileResetImpl implements QProfileReset {
+
+  private final DbClient db;
+  private final QProfileFactory factory;
+  private final RuleActivator activator;
+  private final ActiveRuleIndexer activeRuleIndexer;
+  private final ProfileDefinition[] definitions;
+
+  public QProfileResetImpl(DbClient db, RuleActivator activator, ActiveRuleIndexer activeRuleIndexer, QProfileFactory factory, ProfileDefinition... definitions) {
+    this.db = db;
+    this.activator = activator;
+    this.activeRuleIndexer = activeRuleIndexer;
+    this.factory = factory;
+    this.definitions = definitions;
+  }
+
+  public QProfileResetImpl(DbClient db, RuleActivator activator, ActiveRuleIndexer activeRuleIndexer, QProfileFactory factory) {
+    this(db, activator, activeRuleIndexer, factory, new ProfileDefinition[0]);
+  }
+
+  @Override
+  public void resetLanguage(DbSession dbSession, OrganizationDto organization, String language) {
+    ListMultimap<QProfileName, RulesProfile> profilesByName = loadDefinitionsGroupedByName(language);
+    for (Map.Entry<QProfileName, Collection<RulesProfile>> entry : profilesByName.asMap().entrySet()) {
+      QProfileName profileName = entry.getKey();
+      QualityProfileDto profile = factory.getOrCreate(dbSession, organization, profileName);
+      List<RuleActivation> activations = Lists.newArrayList();
+      for (RulesProfile def : entry.getValue()) {
+        for (ActiveRule activeRule : def.getActiveRules()) {
+          RuleActivation activation = new RuleActivation(RuleKey.of(activeRule.getRepositoryKey(), activeRule.getRuleKey()));
+          activation.setSeverity(activeRule.getSeverity().name());
+          if (!activeRule.getActiveRuleParams().isEmpty()) {
```

Comment:
![CRITICAL](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-critical.png 'Severity: CRITICAL') Refactor this code to not nest more than 3 if/for/while/switch/try statements. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://next.sonarqube.com/sonarqube/coding_rules#rule_key=squid%3AS134)